main.py

```diff
@@ -28,25 +53,51 @@ def main(coco_filepath: file_path, image_library_path: dir_path, input_image_nam
     # Get data
     image_path = os.path.join(image_library_path, input_image_name)
     image = cv2.imread(image_path)
+    cv2.namedWindow("Output image preview")
+    # Create trackbars to set parameters
+    # Get dilated mask function trackbar
+    cv2.createTrackbar("Dilation length","Output image preview", 51, 255, odd_dilation_callback)
+    # Smooth mask function trackbars
+    cv2.createTrackbar("Gaussian blur kernel(smmoth mask)","Output image preview", 75, 555, odd_gaussian_blur_callback)
+    cv2.createTrackbar("Threshold value","Output image preview", 128, 255, empty_callback)
+    cv2.createTrackbar("Max value with thresh binary","Output image preview", 255, 255, empty_callback)
+    # Blurred mask
+    cv2.createTrackbar("Blur length","Output image preview", 149, 255, odd_blurred_mask_callback)
     annotations = json.load(open(coco_filepath))
+    
     # Main loop
     while True:
         # Process mask
         contours, number_of_objects_in_the_photo = get_contours(annotations, input_image_name, object_index)
         mask = get_mask_from_contours(image, contours)
 
         if mask is not None:
-            mask_blurred = process_blurred_mask(mask)
+            # Get dilated mask. Bigger dilation_length - more expanded boundaries around litter image
+            dilation_length = cv2.getTrackbarPos("Dilation length","Output image preview")
+            dilated = get_dilated_mask(mask, dilation_length)
+            # Get smoothed mask. Smaller values of gauusian_blur_kernel will preserve more details, while larger values will result in more smoothing
+            gauusian_blur_kernel = cv2.getTrackbarPos("Gaussian blur kernel(smmoth mask)","Output image preview")
+            # Pixels with intensity values below threshold_value will be set to 0
+            threshold_value = cv2.getTrackbarPos("Threshold value","Output image preview")
+            # For binary thresholding, pixels above the threshold get set to maxvalue
+            maxvalue = cv2.getTrackbarPos("Max value with thresh binary","Output image preview")
+            mask_smooth = smooth_mask(dilated, gauusian_blur_kernel, threshold_value, maxvalue)
+            # The size of the kernel for the Gaussian blur applied to smoothed mask
+            blur_length = cv2.getTrackbarPos("Blur length","Output image preview")
+            mask_blurred = cv2.GaussianBlur(mask_smooth, (blur_length, blur_length), 0)
+            mask_blurred = cv2.cvtColor(mask_blurred, cv2.COLOR_GRAY2BGR)
+
             mask_contour = get_mask_contours(mask_blurred)
             x, y, w, h = cv2.boundingRect(mask_contour)
 
             img_cropped = image[y:y + h, x:x + w]
             mask_cropped = mask_blurred[y:y + h, x:x + w]
 
             if scale is not None:
-                img_cropped = cv2.resize(img_cropped, (0, 0), fx=scale, fy=scale)
```

Comment:
Resize error found: 
```bash
Traceback (most recent call last):
  File "/home/milosz/Praca_magisterska/synthetic_data_generation/main.py", line 234, in <module>
    main(coco_filepath=args.coco, image_library_path=args.library, input_image_name=args.input,
  File "/home/milosz/Praca_magisterska/synthetic_data_generation/main.py", line 134, in main
    output_image_copy[y_offset:y_offset + h, x_offset:x_offset + w] = alpha_blend(
  File "/home/milosz/Praca_magisterska/synthetic_data_generation/utilities/image_processing.py", line 157, in alpha_blend
    mask_resized = cv2.resize(mask, (background.shape[1], background.shape[0]))
cv2.error: OpenCV(4.8.1) /io/opencv/modules/imgproc/src/resize.cpp:4062: error: (-215:Assertion failed) !ssize.empty() in function 'resize'
```
Configuration:
```
--coco
/home/milosz/Praca_magisterska/UAVVaste/annotations/annotations.json
--library
/home/milosz/Praca_magisterska/UAVVaste/images
--input
batch_05_img_1780.jpg
--output
/home/milosz/Praca_magisterska/Data_synthetic/Photos/beach_1.jpg
--method
alpha_blend
```
Moved Gaussian blur kernel trackbar to max val